### DISCUSSION TITLE
[FR] Support to update related settings after renaming a toolbar

### DISCUSSION THREAD

**Discussion by Moyf** [chrisgurney/obsidian-note-toolbar#105] [Ideas]
Hey~ It's me again! 
I hope I'm not being too bothersome 😂

As you can see, we can now configure the "Toolbar name" in several locations, such as in the Folder Mapping:
![image](https://github.com/user-attachments/assets/cd93a808-b981-40d5-9ee7-873f555d3da0)

Or in the Item Menu:
![image](https://github.com/user-attachments/assets/d7230734-af66-45ee-8e2e-daa96989f071)

And when we change the name, we currently need to update the settings manually, one by one.

What do you think about showing a confirm window after users renaming a toolbar and ask " **Do you want to update name in related settings** " ?

(Confirm firstly rather than directly change the name in other settings - because I'm afraid sometimes maybe people would prefer to keep the old setting's name? I'm not very sure 🤔 )

**Comment by chrisgurney** [comment]
@Moyf Thanks for bringing this up!

Yes, I have an item in my list to manage mappings and items (as they relate to toolbars) by their ID, versus by their name. If I can get that working, there's no need to update toolbar names.

_However_, the only one I won't be able to address (easily) is if the name is changed and it's being used in the `notetoolbar` property -- I haven't found a way yet to query properties across all notes in the API.

To help the problem currently, I added the _Usage_ section. I originally had it up near the _Name_ field but thought it wouldn't really help 99% of the time. To address property usage, I added that link to open the search bar for properties; I may have to provide such a shortcut around the _Name_ field again.

**Comment by chrisgurney** [comment]
@Moyf Toolbars now use IDs as of [1.10.0-beta-04](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.10.0-beta-04). This means you can rename a toolbar and references to that toolbar will no longer be invalid.

Thanks again for raising this issue/idea!

**Reply by Moyf** [reply]
Amazing! Glad to hear that 🥳

**Comment by chrisgurney** [comment]
Closing as toolbar IDs are now being used in settings, as of the [1.10.1 release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.10.1).
